### PR TITLE
Ubuntu: add 24.04

### PR DESCRIPTION
**Pull request by moto-timo**
https://releases.ubuntu.com/noble/

                           RELEASED  END OF STANDARD SUPPORT
24.04 LTS (Noble Numbat)   Apr 2024  Apr 2034

https://ubuntu.com/about/release-cycle

https://canonical.com/blog/canonical-releases-ubuntu-24-04-noble-numbat

### CODE REVIEW

**Comment by Ecordonnier** [comment]
Is something missing before this can be merged?

File: dockerfiles/ubuntu/ubuntu-24.04/ubuntu-24.04-base/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+FROM ubuntu:24.04
+
+ARG TARGETPLATFORM
+
+RUN apt-get update && \
+    DEBIAN_FRONTEND=noninteractive apt-get install -y \
+        gawk \
+        wget \
+        git-core \
```
Comment (Ecordonnier) [inline dockerfiles/ubuntu/ubuntu-24.04/ubuntu-24.04-base/Dockerfile:16]: git-core has been renamed to git a long time ago. "git-core" is provided as virtual package only for backwards compatibility.

**Comment by moto-timo** [comment]
> Is something missing before this can be merged?

It failed to build
`2.313 E: Unable to locate package libegl1-mesa `

and the Yocto Project AutoBuilder was also having issues with Ubuntu 24.04 (notice there is no 24.04 [worker](https://autobuilder.yoctoproject.org/typhoon/#/workers), for good reason). I only have limited time to spend on this, and if it does not pass automation it will not be merged.

**Comment by Ecordonnier** [comment]
> > Is something missing before this can be merged?
> 
> It failed to build `2.313 E: Unable to locate package libegl1-mesa `
> 
> and the Yocto Project AutoBuilder was also having issues with Ubuntu 24.04 (notice there is no 24.04 [worker](https://autobuilder.yoctoproject.org/typhoon/#/workers), for good reason). I only have limited time to spend on this, and if it does not pass automation it will not be merged.

Thanks, I was not trying to push you, but rather genuinely trying to understand if something was blocking this.
I'll have a look at the replacement for libegl1-mesa. Also the 24.04 image will need an apparmor profile for bitbake as documented in https://bugs.launchpad.net/ubuntu/+source/apparmor/+bug/2056555/comments/11

**Comment by Ecordonnier** [comment]
In Ubuntu 22.04, libegl1-mesa was a transitional dummy package whose only dependency was libegl1, thus libegl1 is the correct replacement for libegl1-mesa in Ubuntu 24.04:
https://packages.ubuntu.com/jammy/libegl1-mesa

**Comment by Ecordonnier** [comment]
Also note that libegl1 was removed completely from the ubuntu essential packages, so I think not even libegl1 is needed, and libegl1-mesa can be removed completely: https://git.yoctoproject.org/poky/commit/?id=fd1ea6a600a3b069d6cea4153663ea9720d4eafd 

> drop mesa/sdl from essential host packages
>    
>   They used to be required for qemu graphics support, but neither is being
>  for anything anymore.

**Comment by Ecordonnier** [comment]
@moto-timo Thanks!

**Pull request merged**